lib/src/parse/stylesheet.dart

```diff
@@ -1031,20 +1040,21 @@ abstract class StylesheetParser extends Parser {
   /// [start] should point before the `@`. [child] is called to consume any
   /// children that are specifically allowed in the caller's context.
   IfRule _ifRule(LineScannerState start, Statement child()) {
+    whitespace(allowNewlines: true);
     var ifIndentation = currentIndentation;
     var wasInControlDirective = _inControlDirective;
     _inControlDirective = true;
     var condition = _expression();
     var children = this.children(child);
-    whitespaceWithoutComments();
+    whitespaceWithoutComments(allowNewlines: false);
 
     var clauses = [IfClause(condition, children)];
     ElseClause? lastClause;
 
     while (scanElse(ifIndentation)) {
-      whitespace();
+      whitespace(allowNewlines: true);
```

Comment:
We should _not_ allow newlines here, because `@else` on its own is a complete statement.